src/jit/codegenxarch.cpp

```diff
@@ -2470,9 +2470,21 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
     case GT_JTRUE:
         {
             GenTree *cmp = treeNode->gtOp.gtOp1;
+
             assert(cmp->OperIsCompare());
             assert(compiler->compCurBB->bbJumpKind == BBJ_COND);
 
+#if !defined(_TARGET_64BIT_)
+            GenTreePtr op1 = cmp->gtGetOp1();
+
+            // For long compares, we emit special logic
+            if (varTypeIsLong(op1->TypeGet()))
+            {
+                genJTrueLong(cmp);
+                break;
```

Comment:
IMO, these 'breaks' in the middle of the logic are hard to notice and one has to be very careful while making fixes/changes to code.  How about using 'else' block to put rest of the logic so that there is a single break for entire GT_JTRUE?